lib/invisible_captcha/controller_ext.rb

```diff
@@ -12,6 +12,15 @@ def detect_spam(options = {})
       if invisible_captcha?(options)
         on_spam_action(options)
       end
+      if invisible_captcha_threshold?(options)
```

Comment:
I'd go with 2 tests: one per callback. I mean, in the first test, make the timestamp filter trigger and test just this callback. Then, in a second test, make the timestamp filter to pass, and trigger the honeypot filter.